impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/cisupport/CIDetectorHelper.java

```diff
@@ -33,15 +33,13 @@ public final class CIDetectorHelper {
     private CIDetectorHelper() {}
 
     public static List<CIInfo> detectCI() {
-        List<CIDetector> detectors = ServiceLoader.load(CIDetector.class).stream()
+        List<CIInfo> result = new ArrayList<>(ServiceLoader.load(CIDetector.class).stream()
                 .map(ServiceLoader.Provider::get)
-                .toList();
+                .map(CIDetector::detectCI)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .toList());
```

Comment:
https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/stream/Stream.html#toList():
> The returned List is unmodifiable

so not sure about `removeIf` later.

Maybe `Collectors.toCollection(ArrayList::new)`?

Reply:
I think is fine, this happens "on boot", is not some sort of "hot" spot. The reason why I remove "generic" if there are 2 or more is that I assume there is more specific hit (like GH, sets `CI=true` but also sets `GITHUB_RUNNER=true`)... we can refine this later anyway.

Reply:
Also, to keep things simple I don't want to fiddle with order, as other solution would be to test all, and THEN as last, if no hit, "generic"... but that would assume there is no (and never will be) overlap between all CI systems, which IMO we never know and cannot know (future wise).